tests/FakeItEasy.Specs/AssignsOutAndRefParametersSpecs.cs

```diff
@@ -157,5 +157,31 @@ public static void AssignOutAndRefParameters(IHaveARef subject, string outValue)
             "And the out parameter is set to the specified value"
                 .x(() => outValue.Should().Be(99));
         }
+
+        [Scenario]
+        public static void AssignOutAndRefParameterRecordedCall(
+            IHaveARef subject, string refValue)
+        {
+            "Given a fake with a method that has a ref parameter"
+                .x(() => subject = A.Fake<IHaveARef>());
+
+            "When the fake is configured to assign out and ref parameters when argument matches condition"
+                .x(() =>
+                {
+                    refValue = Condition;
+                    A.CallTo(() => subject.MightReturnAKnownValue(ref refValue))
+                        .AssignsOutAndRefParameters(KnownOutput);
+                });
+
+            "And the configured method is called"
+                .x(() => subject.MightReturnAKnownValue(ref refValue));
+
+            "The the assertion that a call with the expected value has happened succeeds"
+                .x(() =>
+                {
+                    string expectedValue = Condition;
+                    A.CallTo(() => subject.MightReturnAKnownValue(ref expectedValue)).MustHaveHappened();
+                });
+        }

```

Comment:
Since the delegate fakes needed an additional change, I think we should have a spec for them too.